modules/spam/text.py

```diff
@@ -323,19 +321,30 @@ def brainfuck(self, update: Update, context: CallbackContext) -> None:
                     else:
                         data[data_ptr] = 0
             elif command == ".":
-                if chr(data[data_ptr]) not in string.printable:  # Gross flemme
-                    result += " "
+                print(f'printing {data[data_ptr]} ({chr(data[data_ptr])})')
+
+                char_code = data[data_ptr]
+                if char_code == 7:  # bell
+                    result += "🔔"
+                elif char_code == 8:  # backspace
+                    result = result[:-1]
+                elif chr(char_code).isprintable():
+                    result += chr(char_code)
                 else:
-                    result += chr(data[data_ptr])
+                    result += "�"
             elif command == ",":
                 if inputs:
                     data[data_ptr] = ord(inputs.pop(0))
                 else:
-                    # Here, we are cheating:
-                    # We consider that the byte immediately next to the ',' instruction
-                    # will be considered the input to be used.
-                    instr_ptr += 1
-                    data[data_ptr] = ord(instr[instr_ptr])
+                    # a) confusing b) if , is the last character, this is an index out of bounds
+                    # # Here, we are cheating:
+                    # # We consider that the byte immediately next to the ',' instruction
+                    # # will be considered the input to be used.
+                    # instr_ptr += 1
+                    # data[data_ptr] = ord(instr[instr_ptr])
+
+                    update.message.reply_text(f"Error at position {instr_ptr}: , expected input but none was provided.")

```

Comment:
```suggestion
                    update.message.reply_text(f"Error at position {instr_ptr}: `,` expected input but none was provided.")

```

Reply:
ok but then i'm gonna add it to the other error message too